### PR TITLE
feat: icky 1091 cart takeover-headless storefront claim authticket

### PR DESCRIPTION
**Pull request by kibo-snehanerekar**

### CODE REVIEW

File: components/layout/AppHeader/Common/KiboHeader.tsx [hunk 1]
```diff
@@ -87,17 +89,28 @@ const HeaderActionArea = (props: HeaderActionAreaProps) => {
...
-            onAccountIconClick={onAccountIconClick}
-          />
-          <AccountRequestIcon
-            onClick={onAccountRequestClick}
-            isElementVisible={false}
-            iconProps={{ fontSize: isHeaderSmall ? 'small' : 'medium' }}
-            buttonText={t('b2b-account-request')}
-          />
+          {!isCSR && isCSR === undefined && (
+            <StoreFinderIcon
```
Comment (kibo-chandradeeptalaha) [inline components/layout/AppHeader/Common/KiboHeader.tsx]: 1. The condition !isCSR should be enough.
2. wrap these components inside a fragment and put the condition on that fragment.

File: components/layout/AppHeader/Common/KiboHeader.tsx [hunk 2]
```diff
@@ -120,9 +133,11 @@ const TopHeader = ({ navLinks }: { navLinks: NavigationLink[] }) => {
           {navLinks?.map((nav, index) => {
             return (
               <Box key={index}>
-                <StyledLink href={nav.link} passHref>
-                  {t(`${nav.text}`)}
-                </StyledLink>
+                {!isCSR && isCSR === undefined && (
```
Comment (kibo-chandradeeptalaha) [inline components/layout/AppHeader/Common/KiboHeader.tsx]: add this check above like 
`!isCSR && navLinks?.map(...`

File: components/layout/AppHeader/Common/KiboHeader.stories.tsx [hunk 3]
```diff
@@ -28,6 +29,7 @@ Praesent commodo cursus magna, vel scelerisque nisl consectetur et.`
           )
           .join('\n')}
       </Box>
+      <StoreFinderIcon></StoreFinderIcon>
```
Comment (kibo-chandradeeptalaha) [inline components/layout/AppHeader/Common/KiboHeader.stories.tsx]: Why are we adding this here ?

File: components/layout/AppHeader/Common/KiboHeader.spec.tsx [hunk 4]
```diff
@@ -116,7 +126,7 @@ describe('[component] KiboHeader component', () => {
     })
   })
 
-  it('should open b2b account request form dialog when user clicks on B2B Account Request link', async () => {
+  xit('should open b2b account request form dialog when user clicks on B2B Account Request link', async () => {
```
Comment (kibo-chandradeeptalaha) [inline components/layout/AppHeader/Common/KiboHeader.spec.tsx]: remove x

File: components/layout/AppHeader/Mobile/MobileHeader.tsx [hunk 5]
```diff
@@ -48,11 +50,13 @@ const MobileHeader = ({ children }: { children?: React.ReactNode }) => {
           </Link>
         </Box>
 
-        <StoreFinderIcon
-          size="medium"
-          mobileIconColor="black"
-          data-testid="mobile-header-store-icon"
-        />
+        {!isCSR && isCSR === undefined && (
```
Comment (kibo-chandradeeptalaha) [inline components/layout/AppHeader/Mobile/MobileHeader.tsx]: !isCSR check should be enough

File: components/layout/HamburgerMenu/HamburgerMenu.tsx [hunk 6]
```diff
@@ -123,16 +128,18 @@ const HamburgerMenu = (props: HamburgerMenuProps) => {
...
-                    <ListItemText
-                      primary={
-                        <Typography variant="body2" color="text.primary">
-                          {t(`${nav.text}`)}
-                        </Typography>
-                      }
-                      onClick={() => handleNavLinks(nav.link)}
-                    />
-                  </ListItem>
+                  {!isCSR && isCSR === undefined && (
```
Comment (kibo-chandradeeptalaha) [inline components/layout/HamburgerMenu/HamburgerMenu.tsx]: add this check above like
`!isCSR && navLinks?.map(...`

File: hooks/queries/cart-takeover/useGetCartTakeover.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  const response = await client.request({
+    document: getOneTimeSecretQuery,
+    variables: secretId,
+  })
+
+  return response?.GetCartTakeover
+}
+
+/**
+ * [Query hook] useGetCartTakeover uses the graphQL query
```
Comment (kibo-chandradeeptalaha) [inline hooks/queries/cart-takeover/useGetCartTakeover.ts]: Please correct this description

File: hooks/queries/cart-takeover/useGetCartTakeover.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+ * @param secretId - string
+ *
+ * @returns 'response?.GetCartTakeover', which contains shipping methods based on quoteId and draft request.
+ */
+
+export const useGetCartTakeover = (secretId: string): UseGetCartTakeoverResponse => {
+  const { data, isLoading, isSuccess } = useQuery({
+    queryKey: cartTakeoverKeys.all,
+    queryFn: () => getOneTimeSceretMethods(secretId),
+    // cacheTime: 0,
```
Comment (kibo-chandradeeptalaha) [inline hooks/queries/cart-takeover/useGetCartTakeover.ts]: remove commented code

**Review by kibo-chandradeeptalaha — CHANGES_REQUESTED** [review]
Please address these comments.

File: components/layout/AppHeader/Common/KiboHeader.tsx (hunk 1 above)
Comment (kibo-snehanerekar) [reply components/layout/AppHeader/Common/KiboHeader.tsx]: Earlier, the app was not respecting the condition !isCSR on local hence added an undefined one, tested on local again without the undefined condition and it's working fine so removed it.
Wrapped the components in one fragment with a single condition

File: components/layout/AppHeader/Common/KiboHeader.tsx (hunk 2 above)
Comment (kibo-snehanerekar) [reply components/layout/AppHeader/Common/KiboHeader.tsx]: done

File: components/layout/AppHeader/Common/KiboHeader.stories.tsx (hunk 3 above)
Comment (kibo-snehanerekar) [reply components/layout/AppHeader/Common/KiboHeader.stories.tsx]: removed

File: components/layout/AppHeader/Common/KiboHeader.spec.tsx (hunk 4 above)
Comment (kibo-snehanerekar) [reply components/layout/AppHeader/Common/KiboHeader.spec.tsx]: removed

File: components/layout/AppHeader/Mobile/MobileHeader.tsx (hunk 5 above)
Comment (kibo-snehanerekar) [reply components/layout/AppHeader/Mobile/MobileHeader.tsx]: done

File: components/layout/HamburgerMenu/HamburgerMenu.tsx (hunk 6 above)
Comment (kibo-snehanerekar) [reply components/layout/HamburgerMenu/HamburgerMenu.tsx]: done

File: hooks/queries/cart-takeover/useGetCartTakeover.ts (hunk 7 above)
Comment (kibo-snehanerekar) [reply hooks/queries/cart-takeover/useGetCartTakeover.ts]: done

File: hooks/queries/cart-takeover/useGetCartTakeover.ts (hunk 8 above)
Comment (kibo-snehanerekar) [reply hooks/queries/cart-takeover/useGetCartTakeover.ts]: done

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Pull request merged**